outline/shadowsocks/config.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shadowsocks
+
+import (
+	"github.com/Jigsaw-Code/outline-ss-server/client"
+	"github.com/eycorsican/go-tun2socks/common/log"
+)
+
+// Config represents a shadowsocks server configuration.
+// Exported via gobind.
+type Config struct {
```

Comment:
I'd say this is a client configuration. Perhaps call it `ClientConfig`?
It's the thing we need to create and configure a Client.
Hopefully we can make it opaque to platform-specific code.